Index: packages.ent
===================================================================
--- packages.ent	(revision 8043)
+++ packages.ent	(working copy)
@@ -299,11 +299,11 @@
 
 <!ENTITY linux-major-version "2.6">
 <!ENTITY linux-minor-version "20">
-<!ENTITY linux-patch-version "4">
+<!ENTITY linux-patch-version "6">
 <!ENTITY linux-version "&linux-major-version;.&linux-minor-version;.&linux-patch-version;">
 <!ENTITY linux-size "42,359 KB">
 <!ENTITY linux-url "&kernel;linux/kernel/v&linux-major-version;/linux-&linux-version;.tar.bz2">
-<!ENTITY linux-md5 "ad6df0431ce9eb91dd558538e71239de">
+<!ENTITY linux-md5 "dc2f9b8c76d75802d39ad2a99de587d9">
 <!ENTITY linux-home "http://www.kernel.org/">
 <!ENTITY linux-ch8-du "310 - 390 MB">
 <!ENTITY linux-ch8-sbu "1.5 - 3.5 SBU">
