diff -Naur linux-2.6.10.orig/mm/filemap.c linux-2.6.10/mm/filemap.c
--- linux-2.6.10.orig/mm/filemap.c	2005-01-28 21:02:32.000000000 -0500
+++ linux-2.6.10/mm/filemap.c	2005-01-28 21:05:56.000000000 -0500
@@ -642,6 +642,8 @@
 	return ret;
 }
 
+EXPORT_SYMBOL(find_get_pages_tag);
+
 /*
  * Same as grab_cache_page, but do not wait if the page is unavailable.
  * This is intended for speculative data generators, where the data can
