diff -urN isoinfo-0.03.02/isoinfo.c isoinfo-0.03.02.fixed/isoinfo.c
--- isoinfo-0.03.02/isoinfo.c	2002-10-13 03:28:17.000000000 +0600
+++ isoinfo-0.03.02.fixed/isoinfo.c	2005-04-22 13:31:33.000000000 +0600
@@ -103,7 +103,7 @@
 iso_str( char* p, int n ) {
   static char string[256];
   int i= 0;
-  while( i < n ) string[i]= p[i++];
+  while( i < n ) { string[i]= p[i]; i++;}
   string[i--]= '\0';
   while( string[i] == ' ' ) string[i--]= '\0';
   return( string );
@@ -340,7 +340,7 @@
   if( in_fd != STDIN_FILENO ) {
     if( (in_fd=open(in_fn,O_RDONLY)) == -1 ) {
       fprintf( ERROR_CHANNEL, "%s: Cannot open `%s'. %s.\n",
-               pn, in_fn, sys_errlist[errno] );
+               pn, in_fn, strerror(errno) );
       retval= RETVAL_ERROR; goto DIE_NOW;
     }
   }
@@ -354,12 +354,12 @@
     if( in_fd != STDIN_FILENO ) { /* we can lseek() */
       if( lseek(in_fd,blk_to_read*ISO_BLOCK_SIZE,SEEK_SET) == -1 ) {
         fprintf( ERROR_CHANNEL, "%s: Cannot lseek `%s'. %s.\n",
-                 pn, in_fn, sys_errlist[errno] );
+                 pn, in_fn, strerror(errno) );
         retval= RETVAL_ERROR; goto DIE_NOW;
       }
       if( (nbr=my_read(in_fd,buf,ISO_BLOCK_SIZE)) == -1 ) {
         fprintf( ERROR_CHANNEL, "%s: Cannot read block %ld of `%s'. %s.\n",
-                 pn, blk, in_fn, sys_errlist[errno] );
+                 pn, blk, in_fn, strerror(errno) );
         retval= RETVAL_ERROR; goto DIE_NOW;
       }
       blk= blk_to_read;
@@ -411,7 +411,7 @@
 
   if( close(in_fd) == -1 ) {
     fprintf( ERROR_CHANNEL, "%s: Cannot close `%s'. %s.\n",
-             pn, in_fn, sys_errlist[errno] );
+             pn, in_fn, strerror(errno) );
     retval= RETVAL_ERROR; goto DIE_NOW;
   }
 
diff -urN isoinfo-0.03.02/isoinfo.h isoinfo-0.03.02.fixed/isoinfo.h
--- isoinfo-0.03.02/isoinfo.h	2002-10-13 03:28:17.000000000 +0600
+++ isoinfo-0.03.02.fixed/isoinfo.h	2005-04-22 11:00:53.000000000 +0600
@@ -5,7 +5,7 @@
 #undef	DEBUG
 
 #include <stdio.h>		/* printf() */
-#include <getopt.h>		/* getopt() */
+/*#include <getopt.h> */	/* getopt() */
 #include <errno.h>		/* error numbers */
 #include <unistd.h>		/* read(), write(), close(), STDIN_FILENO */
 #include <sys/types.h>		/* open() */
