diff -ur blfs-bootscripts-20060910.orig/blfs/sysconfig/network-devices/services/pppoe blfs-bootscripts-20060910/blfs/sysconfig/network-devices/services/pppoe
--- blfs-bootscripts-20060910.orig/blfs/sysconfig/network-devices/services/pppoe	2005-09-10 13:37:55.000000000 +0000
+++ blfs-bootscripts-20060910/blfs/sysconfig/network-devices/services/pppoe	2006-10-27 15:17:49.000000000 +0000
@@ -10,18 +10,19 @@
 
 . /etc/sysconfig/rc
 . $rc_functions
+. $IFCONFIG
 
 case "$2" in
         up)
                 boot_mesg "Bringing up the PPPoE interface..."
-                /usr/sbin/pppoe-start
-                evaluate_retval
+                modprobe pppoe 2>/dev/null
+                udevsettle
+                loadproc -p /var/run/ppp-${1}.pid pppd call pppoe ${1} updetach linkname ${1} ${PPP_OPTS}
         ;;
 
         down)
                 boot_mesg "Bringing down the PPPoE interface..."
-                /usr/sbin/pppoe-stop
-                evaluate_retval
+                killproc -p /var/run/ppp-${1}.pid pppd
         ;;
 
         *)
